src/commands/subcommand.cc

```diff
@@ -40,74 +41,65 @@ namespace env
 ///////////////////////////////////////////////////////////////////
 
 ///////////////////////////////////////////////////////////////////
-namespace
-{
+// SubcommandOptions
+///////////////////////////////////////////////////////////////////
 
-  std::ostream & dumpNameTableOn( std::ostream & str, const std::vector<boost::string_ref> & names_r )
-  {
-    const std::string indent( 2, ' ' );
+namespace	// subcommand detetction
+{///////////////////////////////////////////////////////////////////
 
-    if ( names_r.empty() )
-    {
-      str << indent << "<" << _("none") << ">" << endl;
-    }
-    else
-    {
-      unsigned colw = 0;
-      for ( const auto & el : names_r )
-      {
-	unsigned w = el.size();
-	if ( w > colw ) colw = w;
-      }
+  /** Strictly compare \ref SubcommandOptions::Detected according to _cmd. */
+  struct DetectedCommandsCompare
+  {
+    bool operator()( const SubcommandOptions::Detected & lhs, const SubcommandOptions::Detected & rhs ) const
+    { return lhs._cmd < rhs._cmd; }
+  };
 
-      str::Format fmter( indent+"%-"+str::numstring(colw)+"s" );
-      unsigned maxcols = Zypper::instance().out().defaultFormatWidth() / ( colw + indent.size() );
-      if ( maxcols == 0 ) maxcols = 1;
+  /** First cmd detected shadows later ones. */
+  using DetectedCommands = std::set<SubcommandOptions::Detected, DetectedCommandsCompare>;
 
-#if 1
-      // vertical
-      unsigned maxrows = ( names_r.size() + maxcols - 1 ) / maxcols;
-      maxcols = ( names_r.size() + maxrows - 1 ) / maxrows;
+  /** Command name,summaries for help. */
+  using CommandSummaries = std::map<std::string,std::string>;
 
-      for ( unsigned r = 0; r < maxrows; ++r )
-      {
-	unsigned idx = r;
-	for ( unsigned c = 0; c < maxcols; ++c )
-	{
-	  str << ( fmter % names_r[idx] );
-	  if ( (idx += maxrows) >= names_r.size() )
-	    break;
-	}
-	str << endl;
+  /** Get command summaries for help. */
+  inline CommandSummaries getCommandsummaries( const DetectedCommands & commands_r )
+  {
+    CommandSummaries ret;
+    for ( auto & cmd : commands_r ) {
+      std::string sum { ExternalProgram( "man -f "+cmd._name ).receiveLine() };
+      if ( ! sum.empty() ) {
+	// # man -f zypper
+	// zypper (8)           - Command-line interface to ZYpp system management library (libzypp)
+	static const std::string sep { " - " };
+	std::string::size_type pos = sum.find( sep );
+	if ( pos != std::string::npos )
+	  sum.erase( 0, pos + sep.size() );
       }
-#else
-      // horizontal
-      for ( unsigned idx = 0; idx < names_r.size(); ++idx )
-      {
-	if ( idx && idx % maxcols == 0 )
-	  str << endl;
-	str << ( fmter % names_r[idx] );
+      else {
+	// translators: %1% is the name of the command which has no man page available.
+	static str::Format fmt( "<"+ LOWLIGHTString(_("No manual entry for %1%")).str() + ">" );
```

Comment:
Why is this static? You are changing the variable in the next line, leaving it changed for the next time that code is called. Doesn'T that change the behaviour?

Reply:
No. The format string itself does not change anymore. It can be reused in subsequent substitutions. Just the state of the internal stream object that builds the substituted string changes when placeholder arguments are fed.